opamp/observiq/updater_manager.go

```diff
@@ -14,12 +14,67 @@
 
 package observiq
 
-const updaterDir = "latest"
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"go.uber.org/zap"
+)
 
-var updaterName = "updater"
+const updaterDir = "latest"
 
 // updaterManager handles working with the Updater binary
 type updaterManager interface {
 	// StartAndMonitorUpdater starts the Updater binary and monitors it for failure
 	StartAndMonitorUpdater() error
 }
+
+// copyExecutable copies the executable at the input file path to the cwd.
+// Returns the output path of the executable.
+func copyExecutable(logger *zap.Logger, inputPath, cwd string) (string, error) {
+	inputPathClean := filepath.Clean(inputPath)
+
+	inputFile, err := os.Open(inputPathClean)
+	if err != nil {
+		return "", fmt.Errorf("failed to open updater binary for reading: %w", err)
+	}
+	defer func() {
+		if err := inputFile.Close(); err != nil {
+			logger.Error("Failed to close input file", zap.Error(err))
+		}
+	}()
+
+	// Output path is just whatever the actual file name is (e.g. updater.exe),
+	// on top of the CWD. We take the absolute path, because it is needed to actually ensure you can
+	// exec a file not on your PATH.
+	outputPath, err := filepath.Abs(filepath.Join(cwd, filepath.Base(inputPath)))
+	if err != nil {
+		return "", fmt.Errorf("failed to get absolute path for output: %w", err)
+	}
+
+	outputPathClean := filepath.Clean(outputPath)
+
+	// Remove the file if it already exists, need this for macOS
```

Comment:
@BinaryFissionGames just a note that we "may" not need this if you're just assuming because of that code I added. the problem I was seeing when copying over old files existed on executing the collector in the install directory, but not the updater for whatever reason.

Reply:
Yeah, I kinda assumed from your findings with the collector.

What'll end up happening is the updater will be copied over the previous updater in a similar fashion, so I thought this would also be effected.